dispatcherd/registry.py

```diff
@@ -69,37 +89,58 @@ def get_async_body(
         """
         body = self.publication_defaults()
         # These params are forced to be set on every submission, can not be generic to task
-        body.update({'uuid': uuid or str(uuid4()), 'args': args or [], 'kwargs': kwargs or {}})
-
-        # TODO: callback to add other things, guid in case of AWX
-
-        if bind:  # only needed for testing
+        body['uuid'] = uuid or str(uuid4())
+
+        if args:
+            body['args'] = args
+        if kwargs:
+            body['kwargs'] = kwargs
+        # The bind param in the submission data is only needed for testing
+        # normally this should be applied on the task decorator
+        if bind:
             body['bind'] = bind
-        if on_duplicate:
-            body['on_duplicate'] = on_duplicate
-        if delay:
-            body['delay'] = delay
         if timeout:
             body['timeout'] = timeout
 
+        for part in processor_options:
+            body.update(part.to_dict())
+
         return body
 
-    def apply_async(self, args=None, kwargs=None, queue=None, uuid=None, settings: LazySettings = global_settings, **kw) -> Tuple[dict, str]:
-        queue = queue or self.queue
+    def apply_async(
+        self,
+        args: Optional[tuple] = None,
+        kwargs: Optional[dict] = None,
+        queue: Union[str, Callable[..., str], None] = None,
+        uuid: Optional[str] = None,
+        settings: LazySettings = global_settings,
+        bind: bool = False,
+        timeout: Optional[float] = 0.0,
+        processor_options: Iterable[ProcessorParams] = (),
+    ) -> Tuple[dict, str]:
+        """Submit a task to be ran by dispatcherd worker(s)
+
+        This submission does not provide confirmation.
+        If you need confirmation from the service, look into the "run" control command.
+        """
 
-        if callable(queue):
-            queue = queue()
+        resolved_queue: str
+        if queue is None:
+            resolved_queue = self.queue
```

Comment:
Since `self.queue` has also `None` as default and comes from the caller also with `None` as default, should we check it before the assignment or it is safe? I can not find where we ensure the value of the queue.

Reply:
I wanted to be able to set the default in the config, and the most logical place was in the broker, so the ultimate enforcement has to be there:

https://github.com/ansible/dispatcherd/blob/dc16ffa6b6e8c097f48dde62f521c68982e2477d/dispatcherd/brokers/pg_notify.py#L126-L132

I'll leave a comment saying this. Might have to put that comment in 2 or 3 places.